File: front/src/types/medical-record.ts
export type VitalSigns = {
  temperature: number | null;
  heartRate: number | null;
  bloodPressureSystolic: number | null;
  bloodPressureDiastolic: number | null;
  weightKg: number | null;
};

export type MedicalRecord = {
  id: string;
  appointmentId: string;
  doctorName: string;
  visitDate: string;
  symptoms: string;
  diagnosis: string;
  notes: string | null;
  vitalSigns: VitalSigns | null;
  createdAt: string;
};

File: Backend/Domain/Entities/MedicalRecord.cs
namespace Domain.Entities;

public class MedicalRecord
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid AppointmentId { get; set; }
    public Guid DoctorId { get; set; }
    public Guid PatientId { get; set; }
    public string Symptoms { get; set; } = null!;
    public string Diagnosis { get; set; } = null!;
    public string? Notes { get; set; }
    public VitalSigns? VitalSigns { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation
    public Appointment Appointment { get; set; } = null!;
    public Doctor Doctor { get; set; } = null!;
    public Patient Patient { get; set; } = null!;
}
